File: RealLifeUi/Html/Componente/Login/DivLoginTab.cs
using NetZ.Web.Html;
using NetZ.Web.Html.Componente;
using NetZ.Web.Server.Arquivo.Css;

namespace RealLifeUi.Html.Componente.Login
{
    internal class DivLoginTab : ComponenteHtml
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private Div _divTabContaCadastro;
        private Div _divTabLoginEntrar;

        private Div divTabContaCadastro
        {
            get
            {
                if (_divTabContaCadastro != null)
                {
                    return _divTabContaCadastro;
                }

                _divTabContaCadastro = new Div();

                return _divTabContaCadastro;
            }
        }

        private Div divTabLoginEntrar
        {
            get
            {
                if (_divTabLoginEntrar != null)
                {
                    return _divTabLoginEntrar;
                }

                _divTabLoginEntrar = new Div();

                return _divTabLoginEntrar;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override void inicializar()
        {
            base.inicializar();

            this.strId = this.GetType().Name;

            this.divTabLoginEntrar.strConteudo = "Login";
            this.divTabContaCadastro.strConteudo = "Nova conta";
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            this.divTabLoginEntrar.setPai(this);
            this.divTabContaCadastro.setPai(this);

            new LimiteFloat().setPai(this);
        }

        protected override void setCss(CssArquivoBase css)
        {
            base.setCss(css);

            this.addCss(css.setBackgroundColor("white"));
            this.addCss(css.setBorderLeft());
            this.addCss(css.setBorderRight());
            this.addCss(css.setBorderTop());
            this.addCss(css.setMarginTop(1, "vh"));
            this.addCss(css.setTextAlign("center"));

            this.divTabContaCadastro.addCss(css.setFloat("right"));
            this.divTabContaCadastro.addCss(css.setWidth(50, "%"));

            this.divTabLoginEntrar.addCss(css.setFloat("left"));
            this.divTabLoginEntrar.addCss(css.setFontWeight("bold"));
            this.divTabLoginEntrar.addCss(css.setWidth(50, "%"));
        }

        protected override void setStrId(string strId)
        {
            base.setStrId(strId);

            this.divTabContaCadastro.strId = (strId + "_divTabContaCadastro");
            this.divTabLoginEntrar.strId = (strId + "_divTabLoginEntrar");
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: RealLiteUi.TypeScript/html/componente/login/DivLoginTab.ts
/// <reference path="../../../../Web.TypeScript/erro/Erro.ts"/>
/// <reference path="../../../../Web.TypeScript/html/componente/ComponenteHtml.ts"/>
/// <reference path="../../../../Web.TypeScript/html/Div.ts"/>
/// <reference path="../../../../Web.TypeScript/OnClickListener.ts"/>
/// <reference path="../../../../Web.TypeScript/Utils.ts"/>

module RealLifeUi
{
    // #region Importações

    import ComponenteHtml = Web.ComponenteHtml;
    import Div = Web.Div;
    import Erro = Web.Erro;
    import OnClickListener = Web.OnClickListener;
    import Utils = Web.Utils;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class DivLoginTab extends ComponenteHtml implements OnClickListener
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private _divTabContaCadastro: Div;
        private _divTabLoginEntrar: Div;

        private get divTabContaCadastro(): Div
        {
            if (this._divTabContaCadastro != null)
            {
                return this._divTabContaCadastro;
            }

            this._divTabContaCadastro = new Div(this.strId + "_divTabContaCadastro");

            return this._divTabContaCadastro;
        }

        private get divTabLoginEntrar(): Div
        {
            if (this._divTabLoginEntrar != null)
            {
                return this._divTabLoginEntrar;
            }

            this._divTabLoginEntrar = new Div(this.strId + "_divTabLoginEntrar");

            return this._divTabLoginEntrar;
        }

        // #endregion Atributos

        // #region Construtores

        constructor()
        {
            super(DivLoginTab.name);
        }

        // #endregion Construtores

        // #region Métodos

        private mostrarDivLoginContaCadastro(): void
        {
            this.divTabContaCadastro.jq.css("font-weight", "bold");
            this.divTabLoginEntrar.jq.css("font-weight", "initial");

            PagLogin.i.mostrarDivLoginContaCadastro();
        }

        private mostrarDivLoginEntrar(): void
        {
            this.divTabContaCadastro.jq.css("font-weight", "initial");
            this.divTabLoginEntrar.jq.css("font-weight", "bold");

            PagLogin.i.mostrarDivLoginEntrar();
        }

        protected setEventos(): void
        {
            super.setEventos();

            this.divTabContaCadastro.addEvtOnClickListener(this);
            this.divTabLoginEntrar.addEvtOnClickListener(this);
        }

        // #endregion Métodos

        // #region Eventos

        public onClick(objSender: Object, arg: JQueryEventObject): void
        {
            try
            {
                switch (objSender)
                {
                    case this.divTabContaCadastro:
                        this.mostrarDivLoginContaCadastro();
                        return;

                    case this.divTabLoginEntrar:
                        this.mostrarDivLoginEntrar();
                        return;
                }
            }
            catch (ex)
            {
                new Erro("Erro desconhecido.", ex);
            }
        }

        // #endregion Eventos
    }
}